packages/cloudflare/src/cli/build/bundle-server.ts

```diff
@@ -183,7 +184,7 @@ function createFixRequiresESBuildPlugin(config: Config): Plugin {
     name: "replaceRelative",
     setup(build) {
       // Note: we (empty) shim require-hook modules as they generate problematic code that uses requires
-      build.onResolve({ filter: /^\.(\/|\\)require-hook$/ }, () => ({
+      build.onResolve({ filter: getCrossPlatformPathRegex("^\\./require-hook$", { escape: false }) }, () => ({
```

Comment:
nit: use `String.raw` to avoid double escaping (`\\`)

Reply:
```suggestion
      build.onResolve({ filter: getCrossPlatformPathRegex(String.raw`^\./require-hook$`, { escape: false }) }, () => ({
```